adapters/readpeak/readpeak.go

```diff
@@ -0,0 +1,180 @@
+	bidder := &adapter{
+	  endpoint: config.Endpoint,
+	}
+	return bidder, nil
+}
+
+func (a *adapter) MakeRequests(request *openrtb2.BidRequest, requestInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	var errors []error
+
+	requestCopy := *request
+	var rpExt openrtb_ext.ImpExtReadpeak
+	var imps []openrtb2.Imp
+	for i := 0; i < len(requestCopy.Imp); i++ {
+		if requestCopy.Imp[i].Native == nil && requestCopy.Imp[i].Banner == nil {
+			continue
+		}
+		var impExt adapters.ExtImpBidder
+		err := json.Unmarshal(requestCopy.Imp[i].Ext, &impExt)
+		if err != nil {
+			errors = append(errors, err)
+			continue
+		}
+		err2 := json.Unmarshal(impExt.Bidder, &rpExt)
+		if err2 != nil {
+			errors = append(errors, err2)
+			continue
+		}
+		imp := requestCopy.Imp[i]
+		if rpExt.TagId != "" {
+			imp.TagID = rpExt.TagId
+		}
+		if rpExt.Bidfloor != 0 {
+			imp.BidFloor = rpExt.Bidfloor
+		}
+		imps = append(imps, imp)
+	}
+
+	if len(imps) == 0 {
+		err := &errortypes.BadInput{
+			Message: fmt.Sprintf("Failed to find compatible impressions for request %s", requestCopy.ID),
+		}
+		return nil, []error{err}
+	}
+	requestCopy.Imp = imps
+	publisher := &openrtb2.Publisher{
+		ID: rpExt.PublisherId,
+	}
+
+	if requestCopy.Site != nil {
+		siteCopy := *request.Site
+		if rpExt.SiteId != "" {
+			siteCopy.ID = rpExt.SiteId
+		}
+		siteCopy.Publisher = publisher
+		requestCopy.Site = &siteCopy
+	} else if requestCopy.App != nil {
+		appCopy := *request.App
+		if rpExt.SiteId != "" {
+			appCopy.ID = rpExt.SiteId
+		}
+		appCopy.Publisher = publisher
+		requestCopy.App = &appCopy
+	}
+	
+	requestJSON, err := json.Marshal(requestCopy)
+	if err != nil {
+	  return nil, []error{err}
+	}
+  
+	headers := http.Header{}
+	headers.Add("Content-Type", "application/json;charset=utf-8")
+	headers.Add("Accept", "application/json")
+
+	requestData := &adapters.RequestData{
+	  Method:  "POST",
+	  Uri:     a.endpoint,
+	  Body:    requestJSON,
+	  Headers: headers,
+	}
+  
+	return []*adapters.RequestData{requestData}, errors
+}
+
+func (a *adapter) MakeBids(request *openrtb2.BidRequest, requestData *adapters.RequestData, responseData *adapters.ResponseData) (*adapters.BidderResponse, []error) {
+	if responseData.StatusCode == http.StatusNoContent {
+		return nil, nil
+	}
+
+	if responseData.StatusCode == http.StatusBadRequest {
+		err := &errortypes.BadInput{
+			Message: "Unexpected status code: 400. Bad request from publisher.",
+		}
+		return nil, []error{err}
+	}
+
+	if responseData.StatusCode != http.StatusOK {
+		err := &errortypes.BadServerResponse{
+			Message: fmt.Sprintf("Unexpected status code: %d.", responseData.StatusCode),
+		}
+		return nil, []error{err}
+	}
+
+	var response openrtb2.BidResponse
+	if err := json.Unmarshal(responseData.Body, &response); err != nil {
+		return nil, []error{err}
+	}
+
+	bidResponse := adapters.NewBidderResponseWithBidsCapacity(len(request.Imp))
+	bidResponse.Currency = response.Cur
```

Comment:
by default line 130 will initialize `bidResponse.Currency` as `USD`

reassigning `bidResponse.Currency` on line 131  without checking  `response.Cur` length may overwrite `bidResponse.Currency` with blank value

code block can be refactored as below

```
	bidResponse := adapters.NewBidderResponseWithBidsCapacity(len(request.Imp))
        if len(response.Cur) != 0 {
	    bidResponse.Currency = response.Cur
        }
```

Reply:
Fixed